projects/kit/directives/icon-badge/icon-badge.directive.ts

```diff
@@ -0,0 +1,43 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    inject,
+    Input,
+    signal,
+    ViewEncapsulation,
+} from '@angular/core';
+import {tuiWithStyles} from '@taiga-ui/cdk/utils/miscellaneous';
+import {TuiIcon} from '@taiga-ui/core/components/icon';
+
+@Component({
+    standalone: true,
+    template: '',
+    styleUrls: ['./icon-badge.style.less'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        class: 'tui-icon-badge',
+    },
+})
+class TuiIconBadgeStyles {}
+
+@Directive({
+    standalone: true,
+    selector: 'tui-icon[badge]',
+    host: {
+        tuiIconBadge: '',
+        '[style.--t-icon-badge]': 'badgeSrc()',
```

Comment:
this may cause empty string problem. see https://github.com/taiga-family/taiga-ui/pull/8651